### PR TITLE
Update quarterly roadmap for Q2

### LINKED ISSUE #1971: Prepare to update roadmap for Q2
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
(This section helps Arrow developers understand the context and *why* for this feature, in addition to  the *what*)
Continuing on the same doc as last time, so people have context.   Same as last time, if people fill in information on themselves as an individual ill work on aggregating that into overall Q2 goals.

https://docs.google.com/document/d/1t64vZwZnXm9MyFj2qz3xcAkSxK3Wu12giS3KrS4nDE0/edit

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by matthewmturner**
# Which issue does this PR close?

Closes #1971 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by hntd187** [comment]
I'd like to ideally finalize the implementation for the streaming API and get an experimental impl available via `datafusion-streams` basically requires me to finalize the API contract.

**Comment by jychen7** [comment]
LGTM

**Review by andygrove — APPROVED** [review]
LGTM

**Review by yjshen — APPROVED** [review]

File: docs/source/specification/quarterly_roadmap.md [hunk 1]
```diff
@@ -21,52 +21,65 @@
...
-- Publish official Arrow2 branch
-- Implementation of memory manager (i.e. to enable spilling to disk as needed)
+- IO Improvements
+  - Reading, registering, and writing more file formats from both DataFrame API and SQL
+  - Additional options for IO including partitioning and metadata support
+- Memory Management
+  - Add more operators for memory limited execution
+- Performance
+  - Incorporate row-format into operators such as aggregate
+  - Add row-format benchmarks
```
Comment (Dandandan) [inline docs/source/specification/quarterly_roadmap.md:35]:
```suggestion
  - Add row-format benchmarks
  - Explore JIT-compiling complex expressions
```

File: docs/source/specification/quarterly_roadmap.md [hunk 2]
```diff
@@ -21,52 +21,65 @@
...
 A quarterly roadmap will be published to give the DataFusion community visibility into the priorities of the projects contributors. This roadmap is not binding.
 
-## 2022 Q1
+## 2022 Q2
 
 ### DataFusion Core
 
-- Publish official Arrow2 branch
-- Implementation of memory manager (i.e. to enable spilling to disk as needed)
+- IO Improvements
```
Comment (alamb) [inline docs/source/specification/quarterly_roadmap.md:28]: FYI @tustvold

File: docs/source/specification/quarterly_roadmap.md [hunk 3]
```diff
@@ -21,52 +21,65 @@
...
+  - Reading, registering, and writing more file formats from both DataFrame API and SQL
+  - Additional options for IO including partitioning and metadata support
+- Memory Management
+  - Add more operators for memory limited execution
+- Performance
+  - Incorporate row-format into operators such as aggregate
+  - Add row-format benchmarks
+  - Explore LLVM for JIT, with inline Rust functions as the primary goal
+- Documentation
```
Comment (alamb) [inline docs/source/specification/quarterly_roadmap.md:37]:
```suggestion
  - Improve performance of Sort and Merge using Row Format / JIT expressions
- Documentation
```

Comment (alamb) [reply docs/source/specification/quarterly_roadmap.md:37]: I hope to contribute improvements to the Sort performance (especially for multi-column sorts that include strings) this quarter as well. I don't have any writeup of that yet

**Review by alamb — APPROVED** [review]
Thanks @matthewmturner 

This is going to be an exciting Q2!

File: docs/source/specification/quarterly_roadmap.md (hunk 2 above)
Comment (tustvold) [reply docs/source/specification/quarterly_roadmap.md:28]: Not entirely sure what this specifically is referring to, but I definitely intend to focus on improving the IO and scheduling stories in arrow-rs and DataFusion. See https://github.com/apache/arrow-rs/issues/1473 and https://github.com/apache/arrow-datafusion/issues/2079. Not sure if we want to explicitly call out the scheduling side of this.

I may also get to proper filter pushdown to parquet if I have time - https://github.com/apache/arrow-rs/issues/1191

File: docs/source/specification/quarterly_roadmap.md [hunk 4]
```diff
@@ -21,52 +21,65 @@
...
+## 2022 Q2
 
 ### DataFusion Core
 
-- Publish official Arrow2 branch
-- Implementation of memory manager (i.e. to enable spilling to disk as needed)
+- IO Improvements
+  - Reading, registering, and writing more file formats from both DataFrame API and SQL
+  - Additional options for IO including partitioning and metadata support
+- Memory Management
```
Comment (tustvold) [inline docs/source/specification/quarterly_roadmap.md:31]:
```suggestion
- Work Scheduling
  - Improve predictability, observability and performance of both IO and CPU-bound work
  - Develop a more explicit story for managing parallelism during plan execution
- Memory Management
```

I've yet to create a ticket for this, as I'm still exploring the problem domain, but the precursor discussions can be found https://github.com/apache/arrow-rs/issues/1473 and https://github.com/apache/arrow-datafusion/issues/2079.

File: docs/source/specification/quarterly_roadmap.md (hunk 2 above)
Comment (matthewmturner) [reply docs/source/specification/quarterly_roadmap.md:28]: Thanks @tustvold.  I plan on finishing the work summarized on https://github.com/apache/arrow-datafusion/issues/1777 which is what that refers to

**Comment by matthewmturner** [comment]
thank you @alamb and @tustvold for the suggestions.  I will get them added shortly.

**Comment by alamb** [comment]
Merging and we can keep iterating / updating in follow on PRs if needed.

Thanks again @matthewmturner

**Comment by alamb** [comment]
> I hope to contribute improvements to the Sort performance (especially for multi-column sorts that include strings) this quarter as well. I don't have any writeup of that yet

I filed https://github.com/apache/arrow-datafusion/issues/2427 with some of my thoughts

**Pull request merged**